### PR TITLE
JSAAL-61 basically finised end point for user login

### PR DESCRIPTION
**Pull request by Aaron-Xiaoxingchen**
Please help

### CODE REVIEW

File: src/controller/users.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    return res.sendStatus(400);
+  }
+  try {
+    const [user] = users;
+    if (user.userName === userName && user.passWord === passWord) {
+      const accessTocken = 'little-pony';
+      const { userId } = user;
+      return res.send({ userId, accessTocken });
+    }
+    return res.status(400).send({ message: 'Wrong password' });
```
Comment (ajaxjiang96) [inline src/controller/users.js]:
```suggestion
    return res.sendStatus(401);
```

File: src/controller/users.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  } catch (err) {
+    next(err);
+  }
```
Comment (ajaxjiang96) [inline src/controller/users.js]: no middleware available

**Review by ajaxjiang96 — CHANGES_REQUESTED** [review]

File: src/data/users.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    userId: 1,
+    userName: 'Ajax',
+    passWord: '123456',
+  },
+  {
+    userId: 2,
+    userName: 'Aaron',
+    passWord: '123456',
+  },
+];
```
Comment (huaichaow) [inline src/data/users.js:12]: Let's also add an account for Sherry.

File: src/controller/users.js (hunk 2 above)
Comment (huaichaow) [reply src/controller/users.js]: we can consider using Passport for authentication. http://www.passportjs.org/
It's also a good idea to implement a simple one, it helps to understand the basis of web-based authentication.

File: src/controller/users.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+const { Router } = require('express');
+const users = require('../data/users');
+
+const router = Router();
+
+
```
Comment (ikarasz) [inline src/controller/users.js:11]: no unnecessary empty line

File: src/controller/users.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+// eslint-disable-next-line consistent-return
+async function loginUser(req, res, next) {
+  const { userName, passWord } = req.body;
```
Comment (ikarasz) [inline src/controller/users.js]: username and password should be lowercased

File: src/controller/users.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  if (!userName || !passWord) {
+    return res.sendStatus(400);
+  }
+  try {
+    const [user] = users;
+    if (user.userName === userName && user.passWord === passWord) {
+      const accessTocken = 'little-pony';
+      const { userId } = user;
+      return res.send({ userId, accessTocken });
```
Comment (ikarasz) [inline src/controller/users.js]: do not send the userId with the response

File: src/data/users.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+const users = [
+  {
+    userId: 1,
+    userName: 'Ajax',
+    passWord: '123456',
```
Comment (ikarasz) [inline src/data/users.js:5]: lowercase

File: src/data/users.js [hunk 8]
```diff
@@ -0,0 +1,14 @@
+const users = [
+  {
+    userId: 1,
+    userName: 'Ajax',
```
Comment (ikarasz) [inline src/data/users.js:4]: lowercase

**Review by ikarasz — CHANGES_REQUESTED** [review]

**Review by FairyWind — APPROVED** [review]

**Review by ajaxjiang96 — APPROVED** [review]

**Review by ikarasz — APPROVED** [review]

**Pull request merged**